app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/SearchTest.kt

```diff
@@ -0,0 +1,213 @@
+                ViewMatchers.hasChildCount(dummyDatasets.size)
+            )
+        )
+    }
+
+    @Test
+    fun datasetsOverviewSearchByKeyWordYieldsCorrectResult() {
+        navigateToDatasetsOverview()
+        onView(withId(R.id.action_search)).perform(click(), typeText("kitchen"))
+        onView(withId(R.id.DatasetList_list)).check(
+            ViewAssertions.matches(
+                ViewMatchers.hasChildCount(1)
+            )
+        )
+    }
+
+    @Test
+    fun datasetsOverviewSearchNotFoundYieldsNoResult() {
+        navigateToDatasetsOverview()
+        onView(withId(R.id.action_search)).perform(click(), typeText("hi"))
+        onView(withId(R.id.DatasetList_list)).check(
+            ViewAssertions.matches(
+                ViewMatchers.hasChildCount(0)
+            )
+        )
+    }
+
+    @Test
+    fun datasetsOverviewCanClickOnSubsetOfDatasetsMatchingSearch() {
+        navigateToDatasetsOverview()
+        onView(withId(R.id.action_search)).perform(click(), typeText("kitchen"))
+
+        onView(withId(R.id.DatasetList_list))
+            .perform(
+                RecyclerViewActions.actionOnItemAtPosition<DatasetListRecyclerViewAdapter.ListItemViewHolder>(
+                    0,
+                    click()
+                )
+            )
+        assertCurrentFragmentIsCorrect(R.id.displayDatasetFragment)
+    }
+
+    @Test
+    @Ignore // haven't found a way to clear the text
+    fun datasetsOverviewTypeTextAndThenClearYieldsAllResults() {
+        navigateToDatasetsOverview()
+        onView(withId(R.id.action_search)).perform(click(), typeText("kitchen"), clearText())
+        onView(withId(R.id.DatasetList_list)).check(
+            ViewAssertions.matches(
+                ViewMatchers.hasChildCount(dummyDatasets.size)
+            )
+        )
+    }
+
+    @Test
+    fun learningDatasetSelectionInputEmptyStringYieldsAllResults() {
+        navigateToLearningDatasetSelection()
+        onView(withId(R.id.action_search)).perform(click(), typeText("    "))
+        onView(withId(R.id.DatasetList_list)).check(
+            ViewAssertions.matches(
+                ViewMatchers.hasChildCount(dummyDatasets.size)
+            )
+        )
+    }
+
+    @Test
+    fun learningDatasetSelectionSearchByKeyWordYieldsCorrectResult() {
+        navigateToLearningDatasetSelection()
+        onView(withId(R.id.action_search)).perform(click(), typeText("kitchen"))
+        onView(withId(R.id.DatasetList_list)).check(
+            ViewAssertions.matches(
+                ViewMatchers.hasChildCount(1)
+            )
+        )
+    }
+
+    @Test
+    fun learningDatasetSelectionSearchNotFoundYieldsNoResult() {
+        navigateToLearningDatasetSelection()
+        onView(withId(R.id.action_search)).perform(click(), typeText("hi"))
+        onView(withId(R.id.DatasetList_list)).check(
+            ViewAssertions.matches(
+                ViewMatchers.hasChildCount(0)
+            )
+        )
+    }
```

Comment:
This (and other) tests are copy-pasted, and that's a bad idea. Please refactor this.

Reply:
rookie mistake, thanks